src/coreclr/vm/threadsuspend.cpp

```diff
@@ -1985,7 +1985,9 @@ CONTEXT* AllocateOSContextHelper(BYTE** contextBuffer)
         pfnInitializeContext2(NULL, context, NULL, &contextSize, xStateCompactionMask) :
         InitializeContext(NULL, context, NULL, &contextSize);
 
-    _ASSERTE(!success && GetLastError() == ERROR_INSUFFICIENT_BUFFER);
+    // The following assert is valid, but gets triggered in some Win7 runs with no impact on functionality.
+    // commenting this out to reduce noise, as long as Win7 is supported.
+    // _ASSERTE(!success && GetLastError() == ERROR_INSUFFICIENT_BUFFER);
```

Comment:
do we know what condition fails on win7. Assume that the error returned is not `ERROR_INSUFFICIENT_BUFFER`?

Reply:
I guess it cannot succeed with `NULL` buffer, so most likely it is a different or lost error code.
I do not have Win7 available and as I understand it does not always cause the assert. It is curious, but could be time consuming to find out.

In any case, if this API is not telling us the desired size of the buffer it would be a situation that we cannot handle and it would be fatal in the next call to this API anyways. It would also indicate that something went really wrong on the OS side.

I do not think the value of this assert is very high. I think we use the API correctly and with the comments around it is unlikely to accidentally regress.